internal/vpmem/types.go

```diff
@@ -0,0 +1,122 @@
+package vpmem
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/Microsoft/hcsshim/internal/memory"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const (
+	// MaxVPMEMCount is the maximum number of VPMem devices that may be added to an LCOW
+	// utility VM
+	MaxVPMEMCount = 128
+
+	// DefaultVPMemSizeBytes is the default size of a VPMem device if the create request
+	// doesn't specify.
+	DefaultVPMemSizeBytes = 4 * 1024 * 1024 * 1024 // 4GB
+
+	PageSize = 0x1000
+
+	lcowVPMemPackedLayerMountFmt  = "/run/layers/p%d-%d-%d"
+	lcowVPMemDefaultLayerMountFmt = "/run/layers/p%d"
+
+	vPMemControllerResourceFormat string = "VirtualMachine/Devices/VirtualPMem/Devices/%d"
+	vPMemDeviceResourceFormat     string = "VirtualMachine/Devices/VirtualPMem/Devices/%d/Mappings/%d"
```

Comment:
These are already present in resourcepaths.go, right?

Reply:
right. I was thinking about exporting them, which should be fine I think